cmd/cinder_flexvolume_driver

```diff
@@ -0,0 +1,308 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Notes:
+# This is simple flex volume driver to demostrate use of Cinder static volume
+# for Arktos cluster along with Arktos vm runtime
+# It uses Cinder client and Cinder extended APIs for local attach/detach support
+# package required:
+# JQ
+# python-cinderclient
+# python-brick-cinderclient-ext
+#
+usage() {
+        err "Invalid usage. Usage: "
+        err "\t$0 init"
+        err "\t$0 attach <json params> <nodename>"
+        err "\t$0 detach <mount device> <nodename>"
+        err "\t$0 waitforattach <mount device> <json params>"
+        err "\t$0 mountdevice <mount dir> <mount device> <json params>"
+        err "\t$0 unmountdevice <mount dir>"
+        err "\t$0 isattached <json params> <nodename>"
+        exit 1
+}
+
+err() {
+        echo -ne $* 1>&2
+}
+
+log() {
+        echo -ne $* >&1
+}
+
+logfile() {
+	echo $(date -u +%Y-%m-%dT%H:%M:%SZ) "INFO: " $*  >> /tmp/cinderflexvol_driver.log
+}
+
+#
+# Set the environment variables per your test env
+# Those are used by the cinder client
+#
+setEnv() {
+	export OS_PROJECT_ID=3ab2fbaaf99d422f980c97c4b5092f4b
+    	export OS_USER_DOMAIN_ID=default
+    	export OS_AUTH_URL=http://172.31.14.52/identity
+    	export OS_PASSWORD=Futurewei@123
+    	export OS_USERNAME=admin
+}
+
+getDevice() {
+	VOLUMEID=$1
+	setEnv
+	DMDEV=$(/usr/local/bin/cinder get-volume-paths ${VOLUMEID})
+	echo $(getRealDevice ${DMDEV})
+}
+
+getRealDevice() {
+	readlink -f $1
+}
+
+attach() {
+        logfile "Attach cinder volume to local host" "Args $*"
+        JSON_PARAMS=$1
+        logfile "JSON_PARAMS: " $JSON_PARAMS
+
+	VOLUMEID=$(echo ${JSON_PARAMS} | jq -r '.cinderVolumeID')
+	logfile "Got volumeID: "  ${VOLUMEID}
+
+	VOLUMEPATH=$(echo ${JSON_PARAMS} | jq -r '.path')
+        logfile "Got volumePath: "  ${VOLUMEPATH}
+
+	setEnv
+        /usr/local/bin/cinder local-attach ${VOLUMEID} &> /dev/null
+
+        if [ $? -ne 0 ]; then
+                err "{\"status\": \"Failure\", \"message\": \"Attach OP failed\"}"
+                exit 1
+        fi
+
+	DMDEV=$(getDevice ${VOLUMEID})
+	if [ "${DMDEV}" == "" ]; then
+		exit 1
+	fi
+
+        log "{\"status\": \"Success\", \"device\":\"${DMDEV}\"}"
+	exit 0
+}
+
+detach() {
+	logfile "Detach volume" "Args $*"
+        setEnv
+
+	# For a non-demo implementation, the driver usually keep a map for the volumes and devices
+	# and retrieve the info from the map
+	#
+	# TODO: get the volume from the get-all-volume-paths call
+	#
+	VOLUMEID="3ea0a79e-8ebc-4690-a035-7816c6e7159b"
+	/usr/local/bin/cinder local-detach ${VOLUMEID} &> /dev/null
+	if [ $? -ne 0 ]; then
+		err "{ \"status\": \"Failed\", \"message\": \"Failed to detach volume ${VOLUMEID}\"}"
+                exit 1
+        fi
+
+        log "{\"status\": \"Success\"}"
+	exit 0
+
+}
+
+waitforattach() {
+        logfile "WaitForAttach" "Args S*"
```

Comment:
A general comment: can you help align these indents? It happens in several places in this file.

Reply:
thanks for catching this. will do.

Reply:
thanks for catching this. will do.

Reply:
the VI editor shows aligned and the Code review site shows difference between TAB and 8 spaces... anyways, i enforce TAB in this doc.